### PR TITLE
Add `size_hint` and `step_count` to `PreviousIterator`

### PR DESCRIPTION
**Pull request by matheus23**
Also, a bunch of proptests to verify the behaviour of that.

### CODE REVIEW

File: src/ratchet.rs [hunk 1]
```diff
@@ -423,6 +443,33 @@ impl PreviousIterator {
...
+    /// let mut new_ratchet = old_ratchet.clone();
+    /// new_ratchet.inc_by(100_000);
+    ///
+    /// let iterator = PreviousIterator::new(&old_ratchet, &new_ratchet, 1_000_000_000).unwrap();
+    ///
+    /// assert_eq!(iterator.step_count(), 100_000);
+    /// ```
+    pub fn step_count(&self) -> usize {
+        match self.large_skips.first() {
+            None => 0,
```
[the remark this answers is not in the document]
Comment (appcypher) [reply src/ratchet.rs:464]: > The second half of the tuple that is returned is an [Option](https://doc.rust-lang.org/std/option/enum.Option.html)<[usize](https://doc.rust-lang.org/std/primitive.usize.html)>. A [None](https://doc.rust-lang.org/std/option/enum.Option.html#variant.None) here means that either there is no known upper bound, or the upper bound is larger than [usize](https://doc.rust-lang.org/std/primitive.usize.html).

[the remark this answers is not in the document]
Comment (matheus23) [reply src/ratchet.rs:464]: Great catch. I added that as a regression test & also added another proptest that covers this.
Working on fixing this issue :+1:

[the remark this answers is not in the document]
Comment (matheus23) [reply src/ratchet.rs:464]: Ignore all the size hint stuff. `step_count` is meant to be *exact*. I just implemented `size_hint` because... I could :P

**Comment by matheus23** [comment]
So the fixed implementation for `step_count` works, but I really don't like it.
I tried to keep it O(1) and avoid using the `Ratchet::compare` function, because I was confident that's possible in O(1), and it is, but it looks very ugly.

I think we could perhaps improve the `PreviousIterator` by coming up and being stricter about invariants. But maybe that's a case for another refactor.

File: src/tests.rs [hunk 2]
```diff
@@ -275,6 +278,27 @@ fn test_ratchet_iterator() {
...
+    let mut new_ratchet = old_ratchet.clone();
+    new_ratchet.inc_by(LARGE_EPOCH_LENGTH + 10);
+
+    let mut iterator = PreviousIterator::new(&old_ratchet, &new_ratchet, 1_000_000_000).unwrap();
+
+    for _ in 0..LARGE_EPOCH_LENGTH {
+        assert!(iterator.next().is_some());
+    }
+
+    assert_ne!(iterator.step_count(), 0); // Fails, even though there are 10 steps left.
```
Comment (appcypher) [inline src/tests.rs:293]: Comment no longer applies.

**Review by appcypher — APPROVED** [review]
We can refactor certain aspects about the iterator later. 
But let's ship this!

**Pull request merged**